openfl/federated/task/runner_xgb.py

```diff
@@ -0,0 +1,382 @@
+# Copyright 2020-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+"""XGBoostTaskRunner module."""
+
+# from copy import deepcopy
+# from typing import Iterator, Tuple
+
+import json
+
+import numpy as np
+import xgboost as xgb
+from sklearn.metrics import accuracy_score
+
+from openfl.federated.task.runner import TaskRunner
+from openfl.utilities import Metric, TensorKey, change_tags
+from openfl.utilities.split import split_tensor_dict_for_holdouts
+
+
+class XGBoostTaskRunner(TaskRunner):
+    def __init__(self, **kwargs):
+        """Initializes the XGBoostTaskRunner object.
+
+        Args:
+            **kwargs: Additional parameters to pass to the functions.
+        """
+        super().__init__(**kwargs)
+        self.global_model = None
+        self.required_tensorkeys_for_function = {}
+        self.training_round_completed = False
+
+    def rebuild_model(self, input_tensor_dict):
+        """
+        Rebuilds the model using the provided input tensor dictionary.
+
+        This method checks if the 'local_tree' key in the input tensor dictionary is either a non-empty numpy array
+        or a non-None value. If this condition is met, it updates the internal tensor dictionary with the provided input.
```

Comment:
Just curious, what type could be this "non-None" value, other than a numpy array?

Reply:
This is vestigial. I removed it. Originally, I was trying to send metadata for the fedbagging algorithm as a dictionary alongside the model itself (within the same key), but that ended up not working out